java-frontend/src/main/java/org/sonar/plugins/java/api/tree/Tree.java

```diff
@@ -741,6 +741,14 @@ enum Kind implements GrammarRuleKey {
     @Deprecated(since = "7.7", forRemoval = false)
     DEFAULT_PATTERN(DefaultPatternTree.class),
 
+    /**
+     * {@link RecordPatternTree}
+     *
+     * @since Java 19
+     * @deprecated Preview feature
+     */
+    @Deprecated(since = "7.16", forRemoval = false)
+    RECORD_PATTERN(RecordPatternTree.class),
```

Comment:
Since you added a new tree, you need to also update the `baseTreeVisitor` to give an access to it.
Also, don't forget to modify `visitPatternInstanceOf` to replace the use of `scan(tree.variable());` by `scan(tree.pattern());`